netbox/utilities/forms/forms.py

```diff
@@ -226,6 +230,13 @@ def _clean_csv(self, data):
         reader = csv.reader(stream)
         headers, records = parse_csv(reader)
 
+        if self.model_form:
+            form_fields = self.model_form().fields
+            required_fields = [
+                name for name, field in form_fields.items() if field.required
+            ]
+            validate_csv(headers, form_fields, required_fields)
```

Comment:
Rather than passing the model form to the parent and instantiating it separately, can we just call `validate_csv()` in the model form itself?

Reply:
@jeremystretch  Not sure if I completely follow - the model_form is called/saved per line-item of the CSV, I didn't want to do this validation per line-item as it is really validating just he header line - but the fields need to be pulled from the model_form which I think need to be instantiated to access fields.  I could move the field checking into function in CSVModelForm but would still need to call it here...  Am I missing something obvious here?

Reply:
IIRC validation will fail on the first item if there's an invalid header, so there shouldn't be any penalty for evaluating it per-object (though I need to double-check this).